sigstore-java/src/main/java/dev/sigstore/tuf/HttpMetaFetcher.java

```diff
@@ -58,15 +60,22 @@ public Optional<MetaFetchResult<Root>> getRootAtVersion(int version)
   @Override
   public <T extends SignedTufMeta> Optional<MetaFetchResult<T>> getMeta(Role.Name role, Class<T> t)
       throws IOException, FileExceedsMaxLengthException {
-    return getMeta(role, t, null);
+    return getMeta(getFileName(role, null), t, null);
   }
 
   @Override
   public <T extends SignedTufMeta> Optional<MetaFetchResult<T>> getMeta(
-      Role.Name role, Class<T> t, Integer maxSize)
+      Role.Name role, int version, Class<T> t, Integer maxSize)
       throws IOException, FileExceedsMaxLengthException {
-    String fileName = role.name().toLowerCase(Locale.ROOT) + ".json";
-    return getMeta(fileName, t, maxSize);
+    Preconditions.checkArgument(version > 0, version + " should be a positive value.");
```

Comment:
Prefer template-based API to avoid string concatenation when the check passes: https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkArgument(boolean,%20java.lang.String,%20java.lang.Object...)

```suggestion
    Preconditions.checkArgument(version > 0, "version should be positive, got: %d", version);
```

Reply:
good call. fixed.